### ISSUE TITLE
Relative small PDFs take very long to convert due to "Optimize"

### ISSUE THREAD

**Issue by gentledepp** [dmester/pdftosvg.net#11]
We received two sample PDF files from a customer. They claimed that the "loading of the pdf" was incredibly slow.
And this turned out to be true:
![image](https://github.com/dmester/pdftosvg.net/assets/1175751/8b97998f-fb59-4298-bcbc-40efe717994a)

It appears that "Optimize" tries to remove too many XML nodes. As you can see, "RemoveNode" takes 97,6% of the time.

We will try to investigate ourselves. However, it would be nice if you could give us a hint :-)

[The PDFs can be found here](http://gofile.me/2itnX/OID52ThB8)

Just drop them into the 3rdParty folder and run the ConvertSync test and you'll see.

**Comment by gentledepp** [comment]
Having a quick debug session, it turns out that the input PDF creates a huge SVG, which contains a lot of optimizable attributes.
I.e. attributes which are inheritable are duplicated in the xml tree.

💡Idea: Wouldn't it be better to mainain a stack of inheritable properties and values **during** the conversion so that superfluous attributes aren't created in the first place?

If you aggree: can you give us a hint where we could start to implement this?
My wild guess: Adding stack functionality to `PdfToSvg.Drawing.Handler` ?

**Comment by dmester** [comment]
Thanks for the report! I have a solution that probably work, but I need to verify a bit more. I will publish a new release in a couple of days if it turns out to work ok. You can test it in the branch [issue-11](https://github.com/dmester/pdftosvg.net/tree/issue-11).

Moving the optimization to be done on-the-fly during conversion would indeed technically be a nicer solution, but I think it would be a lot of work, and the performance overhead of the optimization is now more or less negligble compared to no optimization.

**Comment by dmester** [comment]
There is now a new patch (0.14.1) available on NuGet with improved performance. However, the resulting file is still as large as before. I tried some optimizations of the paths to make the SVG smaller, but the effect was pretty small so those changes are not included in this patch. Moving shape attributes to CSS would probably reduce the filesize, but would likely make the rendering performance worse.

I have added it to the list of future improvements to look at further improvements.

Thanks again for reporting problematic PDFs!
